registry/api/v2/urls.go

```diff
@@ -20,6 +21,16 @@ type URLBuilder struct {
 	router *mux.Router
 }
 
+var routeDescriptorsMap = initRouteDescriptorMap()
```

Comment:
Just declare this:

```
// routeNameToDescriptor maps route names to their descriptors. 
var routeNameToDescriptor map[string]RouteDescriptor
````

Then initialize it in the `init` function, in descriptors. That function is special in that it is run on startup.

This should also probably be in descriptors.go, rather than urls.go.